src/kernel.cpp

```diff
@@ -373,7 +373,18 @@ bool ComputeNextStakeModifier(const CBlockIndex* pindexPrev, uint64_t& nStakeMod
 // Get stake modifier checksum
 unsigned int GetStakeModifierChecksum(const CBlockIndex* pindex)
 {
-    assert (pindex->pprev || pindex->GetBlockHash() == (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet));
+    if (pindex->pprev == nullptr) 
+    {
+        if(pindex->GetBlockHash() != (!fTestNet ? hashGenesisBlock : hashGenesisBlockTestNet)) 
+        {
+            //Error on non-genesis blocks that don't have a previous block
+            throw std::runtime_error(
+                "Error: blockchain data corrupted.\n" 
+                "Go to the wallet's data directory and delete the folder txleveldb and the files blk000x.dat (x is any number).\n" 
+                "This requires you to sync again and will temporaily show a balance of 0 GRC\n"
```

Comment:
typo... temporarily.

Let's go with "This requires you to sync again from the beginning and your wallet will temporarily show a balance of 0 GRC."

Reply:
For some reason my spell checker thought that was a valid word and didn't flag it. My ability to make typos is very impressive. Fixed it now